boa/src/context.rs

```diff
@@ -283,7 +283,7 @@ impl Context {
     }
 
     /// Constructs a `RangeError` with the specified message.
-    pub fn construct_range_error<M>(&mut self, message: M) -> Value
+    pub fn construct_range_error<M>(&mut self, message: M) -> Result<Value>
```

Comment:
I think it would be better if we returned Value (like we did before) and and in the body we will call `.expect()` instead of `.expect_err()` since the construction should not fail it makes sense to return `Value`, instead of `Result<Value>`

Reply:
I can see that `construct_range_error` is only ever called with valid messages, is that always the case for all `construct_$TYPE_error` functions?

Reply:
Now that I think about it, the `construct_$TYPE_error` messages could never fail because their message will be converted to `String` so it will always be a valid message. I'll fix it ASAP.

Reply:
> I can see that `construct_range_error` is only ever called with valid messages, is that always the case for all `construct_$TYPE_error` functions?

Yes. they have pretty much the same constructor, so this should not fail